cdr/cdr_pgsql.c

```diff
@@ -783,12 +783,23 @@ static int config_module(int reload)
 
 static int load_module(void)
 {
-	ast_cli_register_multiple(cdr_pgsql_status_cli, sizeof(cdr_pgsql_status_cli) / sizeof(struct ast_cli_entry));
+	int res = AST_MODULE_LOAD_SUCCESS;
+
 	if (config_module(0)) {
-		return AST_MODULE_LOAD_DECLINE;
+		res = AST_MODULE_LOAD_DECLINE;
+	} else {
+		if (ast_cdr_register(name, ast_module_info->description, pgsql_log)) {
+			res = AST_MODULE_LOAD_DECLINE;
+		} else {
+			ast_cli_register_multiple(cdr_pgsql_status_cli, ARRAY_LEN(cdr_pgsql_status_cli));
+		}
 	}
-	return ast_cdr_register(name, ast_module_info->description, pgsql_log)
-		? AST_MODULE_LOAD_DECLINE : 0;
+
+	if (res != AST_MODULE_LOAD_SUCCESS) {
+		unload_module();
+	}
+
+	return res;
```

Comment:
This is a _little_ clunky. Maybe something like:

```
if (config_module(0) || ast_cdr_register(name, ast_module_info->description, pgsql_log)) {
    unload_module();
    return AST_MODULE_LOAD_DECLINE;
}

ast_cli_register_multiple(cdr_pgsql_status_cli, ARRAY_LEN(cdr_pgsql_status_cli));
return AST_MODULE_LOAD_SUCCESS;
```

Reply:
Erring on side of verbosity, but a little less clunkier, thanks!